### DISCUSSION TITLE
2-branch model doesn't work

### DISCUSSION THREAD

**Discussion by stsp** [dosemu2/dosemu2#2165] [General]
It appears the master branch is
not even buildable, as there is no
corresponding fdpp.
I guess the only viable dev model
for dosemu2 is to create the topic
branches and commit them to master
via PRs.
I guess this is what @andrewbird 
asked me to do many times. :)
But I didn't realize we have master
completely broken...

So just merge everything to master
and kill devel branch?

**Comment by andrewbird** [comment]
I think the two branch model used to work okay before fdpp, but the tight coupling between the two projects probably makes it too difficult.  I don't think I've ever used the master branch myself for anything, was it aimed at packagers?

**Comment by stsp** [comment]
> I think the two branch model used to work okay before fdpp

Not quite: it was not updated
for 5-7 years, letting people think
the project is dead.

> I don't think I've ever used the master branch myself for anything, was it aimed at packagers?

IIRC Bart added it (something like
15 years ago) when I pushed some
buggy patch so that he couldn't even
use mouse in dosemu1 any more.
So he asked contributors to use devel,
with an intention to manually cherry-pick
later.
He actually and stubbornly tried to
review+cherry-pick to master for a few
years, but that was quite a horrible task
(sometimes also bug-prone), so eventually
he gave up and started to merge entire
devel from time to time, if it wasn't too
broken (it usually was - no CI).
I think it made sense back when multiple
people had a direct write access to
the svn repo, and the only way to test
their changes was for the maintainer
to manually run programs and see.
These days we have no direct write
access for most people, and I myself
should also start routing all my patches
via CI. Yes, I am super lazy at that, and
CI usually fails for unrelated reasons,
so that would add a trouble or two,
but I understand the need.

In that case devel seems unneeded.

**Comment by stsp** [comment]
> I don't think I've ever used the master branch myself for anything, was it aimed at packagers?

Sorry, I misread the question and
answered why "devel" was created.
But the question is: why do we need
master? I think it was intended for
end-users and maybe packagers,
but it never worked for dosemu2.

So I am going to delete "master"
branch. People got used to "devel"
and moving things to "master" out
of nothing to do, would just be yet
another punishment to those who
still try to build dosemu2 from sources
(and I think they got enough already).

So unless someone thinks otherwise,
"master" will soon be deleted.

**Comment by stsp** [comment]
"master" deleted.

**Comment by jschwartzenberg** [comment]
I think generally there's not really a versioning strategy that makes it easy to find matching versions if you're not using the latest of both dosemu2 and fdpp.

**Reply by stsp** [reply]
... and now also dj64dev.
The only possibility is to use submodules.
But this is also not trouble-free, i.e. what
to do if the different version of any
submodule is already installed system-wide...

**Reply by jschwartzenberg** [reply]
I would say the submodule should have precedence in that case. But maybe that's hard to implement?

**Comment by stsp** [comment]
Среда, 31 июля 2024 г получено от Julius Schwartzenberg:
> I would say the submodule should have precedence in that case. But maybe that's hard to implement?

This will mean completely ignore the
installed packages always. Which
means they haven't had been packaged
separately at all. Which means making
them a part of dosemu2, w/o any
hopes to get them into debian
beforehand. :)

**Reply by stsp** [reply]
Its like even though I am perfectly
aware no one will ever use fdpp/dj64dev
outside of dosemu2, at least I should
pretend its possible. :)
Making them a submodules of dosemu2,
w/o any external use-case, basically
means to stop pretending they are
ever viable outside of dosemu2.
